sbi/inference/base.py

```diff
@@ -152,6 +152,69 @@ def __init__(
             median_observation_distances=[], epochs=[], best_validation_log_probs=[],
         )
 
+    def provide_presimulated(
+        self, theta: Tensor, x: Tensor, from_round: int = 0
+    ) -> None:
+        r"""
+        Provide external $\theta$ and $x$ to be used for training later on.
+
+        Args:
+            theta: Parameter sets used to generate presimulated data.
+            x: Simulation outputs of presimulated data.
+            from_round: Which round the data was simulated from. `from_round=0` means
+                that the data came from the first round.
+        """
+        self._append_to_round_bank(theta, x, from_round)
+
+    def _append_to_round_bank(self, theta: Tensor, x: Tensor, round_: int) -> None:
+        r"""
+        Store data in as entries in a list for each type of variable (parameter/data).
+
+        Stores $\theta$, $x$, prior_masks (indicating if simulations are coming from the
+        prior or not) and a index indicating which round the batch of simulations came
+        from.
+        """
+
+        self._theta_roundwise.append(theta)
+        self._x_roundwise.append(x)
+        self._prior_masks.append(self._mask_sims_from_prior(round_, theta.size(0)))
+        self._data_round_index.append(round_)
+
+    def _get_from_round_bank(
+        self, starting_round: int = 0, exclude_invalid_x: bool = True
+    ) -> Tuple[Tensor, Tensor, Tensor]:
+        r"""
+        Returns all $\theta$, $x$, and prior_masks from rounds >= `starting_round`.
+
+        If requested, do not return invalid data.
+
+        Args:
+            starting_round: The earliest round to return samples from (we start counting
+                from zero).
+            exclude_invalid_x: Whether to exclude simulation outputs `x=NaN` or `x=±∞`
+                during training.
+
+        Returns: Parameters, simulation outputs, prior masks.
+        """
+
+        theta = self._get_data_after_round(self._theta_roundwise, starting_round)
+        x = self._get_data_after_round(self._x_roundwise, starting_round)
+        prior_masks = self._get_data_after_round(self._prior_masks, starting_round)
```

Comment:
I'd suggest getting rid of the new `_get_data_after_round` method and just use `torch.cat` directly, makes it easier to see what is going on and no new method is needed

Reply:
Gets a bit ugly with black. (Each line turns into 3 lines). Also, `_get_data_after_round` does more than just `torch.cat`